### PR TITLE
[Mac] Fix so that when menu is popped up, associated row and control is fo…

### LINKED ISSUE #598: [Mac] PopOvers switch focus back to the last focused control, rather than the current row's controls
Steps to Repro:
Once the PopOver loses focus, it will jump back to the last focused control. Eg. If Filter Control has focus and you then the user opens the `CustomExpression` PopOver, then backs out, it will jump back to the Filter text box. This behaviour happens with all the PopOvers, like the Colour one too etc.

What we hope should happen:
- [ ] If you click a `PropertyButton`, it should first focus that row’s controls, before the PopOver appears, so that should you cancel/accept it it focuses on that row’s control?
- [ ] If you Select a Color and a PopOver appears it should first focus that row’s controls, before the PopOver appears, so that should you cancel/accept it it focuses on that row’s control?

### PR DESCRIPTION
**Pull request by CartBlanche**
…cused.

Fixes #598

### CODE REVIEW

File: Xamarin.PropertyEditing.Mac/Controls/Custom/PropertyButton.cs [hunk 1]
```diff
@@ -142,53 +142,71 @@ private void PopUpContextMenu ()
...
 			var popupMenuEvent = NSEvent.MouseEvent (NSEventType.LeftMouseDown, menuOrigin, (NSEventModifierMask)0x100, 0, this.Window.WindowNumber, this.Window.GraphicsContext, 0, 1, 1);
 
 			NSMenu.PopUpContextMenu (popUpContextMenu, popupMenuEvent, this);
 		}
 
+		private void FocusClickedRow ()
+		{
+			if (Superview != null
+				&& (Superview.Subviews.Length > 1)) {
+				NSView view = Superview.Subviews[1];
```
Comment (ermau) [inline Xamarin.PropertyEditing.Mac/Controls/Custom/PropertyButton.cs]: This seems very fragile, why don't we search the subviews for a `PropertyEditorControl` instead?

Comment (CartBlanche) [reply Xamarin.PropertyEditing.Mac/Controls/Custom/PropertyButton.cs]: We can.

Comment (CartBlanche) [reply Xamarin.PropertyEditing.Mac/Controls/Custom/PropertyButton.cs]: Found an issue where it wasn't focusing all the rows. A more recursive solution seems to work.

File: Xamarin.PropertyEditing.Mac/Controls/Custom/PropertyButton.cs [hunk 2]
```diff
@@ -142,53 +142,76 @@ private void PopUpContextMenu ()
...
 		}
 
+		private void FocusClickedRow ()
+		{
+			if (Superview != null) {
+				FindFocusableKeyView (Superview.Subviews);
+			}
+		}
+
+		private void FindFocusableKeyView (NSView[] subViews)
```
[the remark this answers is not in the document]
Comment (CartBlanche) [reply Xamarin.PropertyEditing.Mac/Controls/Custom/PropertyButton.cs]: Agreed. Fixed.

File: Xamarin.PropertyEditing.Mac/Controls/Custom/PropertyButton.cs [hunk 3]
```diff
@@ -142,53 +142,76 @@ private void PopUpContextMenu ()
...
 				
 				switch (this.viewModel.ValueSource) {
-				case ValueSource.Binding:
-					Image = focused ? this.hostResources.GetNamedImage ("pe-property-button-bound-mac-active-10") : this.hostResources.GetNamedImage ("pe-property-button-bound-mac-10");
-					break;
 
-				case ValueSource.Default:
-					Image = focused ? this.hostResources.GetNamedImage ("pe-property-button-default-mac-active-10") : this.hostResources.GetNamedImage ("pe-property-button-default-mac-10");
-					break;
+					case ValueSource.Binding:
```
[the remark this answers is not in the document]
Comment (CartBlanche) [reply Xamarin.PropertyEditing.Mac/Controls/Custom/PropertyButton.cs]: Fixed.

**Comment by ermau** [comment]
This doesn't actually fix the second element of #598.

**Pull request merged**